### PR TITLE
xml (full)diff target enhancements

### PR DESCRIPTION
**Pull request by jnpkrn**

### CODE REVIEW

File: xml/Makefile.am [hunk 1]
```diff
@@ -106,39 +106,63 @@ pacemaker-%.rng: $(RNG_files) best-match.sh Makefile.am
 	echo "<grammar xmlns='http://relaxng.org/ns/structure/1.0' datatypeLibrary='http://www.w3.org/2001/XMLSchema-datatypes'>" >> $@
 	echo "  <start>" >> $@
 	echo "    <element name='cib'>" >> $@
-	$(top_srcdir)/xml/best-match.sh cib $(*) $(@) "      "
+	./best-match.sh cib $(*) $(@) "      "
```
Comment (beekhof) [inline xml/Makefile.am]: I think that breaks when you build from outside of the source tree.
Which people like to do apparently

Comment (jnpkrn) [inline xml/Makefile.am]: On 09/02/16 17:03 -0800, Andrew Beekhof wrote:
>> @@ -106,39 +106,63 @@ pacemaker-%.rng: $(RNG_files) best-match.sh Makefile.am
>>  	echo "<grammar xmlns='http://relaxng.org/ns/structure/1.0' datatypeLibrary='http://www.w3.org/2001/XMLSchema-datatypes'>" >> $@
>>  	echo "  <start>" >> $@
>>  	echo "    <element name='cib'>" >> $@
>> -	$(top_srcdir)/xml/best-match.sh cib $(*) $(@) "      "
>> +	./best-match.sh cib $(*) $(@) "      "
> 
> I think that breaks when you build from outside of the source tree.
> Which people like to do apparently

Honestly, I cannot think of any use case that wouldn't break before
and is broken after this change.  In fact, the original usage was
even more fragile as you couldn't use Makefile* and best-match.sh
under any custom location, there would have to be a parental dir
hierarchy with a traversal containing "xml" directory at the very
least, which is not strictly needed now.

-- 
Jan (Poki)

Comment (kgaillot) [inline xml/Makefile.am]: Right, top_srcdir can be relative ("..") so I don't see any benefit in how it was done before. Perhaps ${srcdir}/best-match.sh would be more automake-y but probably not any better.

**Comment by kgaillot** [comment]
I'd prefer to put the new documentation in Readme.md rather than its own file, and get rid of the unnecessary target dependencies, but this is good to go.

**Pull request merged**